client/me/purchases/cancel-purchase/index.jsx

```diff
@@ -209,49 +229,113 @@ class CancelPurchase extends Component {
-					<div className="cancel-purchase__refund-amount">
-						{ this.renderFooterText( this.props ) }
+					<Gridicon icon="chevron-left" size={ 18 } />
+					{ this.props.translate( 'Back' ) }
+				</Button>
+
+				<FormattedHeader
+					className="cancel-purchase__formatter-header"
+					brandFont
+					headerText={ heading }
+					align="left"
+				/>
+
+				<div className="cancel-purchase__layout">
+					<div className="cancel-purchase__layout-col cancel-purchase__layout-col-left">
+						<PurchaseSiteHeader siteId={ siteId } name={ siteName } domain={ siteDomain } />
+
+						<CancelPurchaseRefundInformation
+							purchase={ purchase }
+							isJetpackPurchase={ isJetpackPurchase }
+							includedDomainPurchase={ this.props.includedDomainPurchase }
+							confirmBundledDomain={ this.state.confirmCancelBundledDomain }
+							cancelBundledDomain={ this.state.cancelBundledDomain }
+							onCancelConfirmationStateChange={ this.onCancelConfirmationStateChange }
+							site={ this.props.site }
+						/>
+
+						{ isSubscription( purchase ) && (
+							<Fragment>
+								<div className="purchase-cancel-buttons">
+									<CancelPurchaseButton
+										purchase={ purchase }
+										includedDomainPurchase={ this.props.includedDomainPurchase }
+										disabled={
+											this.state.cancelBundledDomain && ! this.state.confirmCancelBundledDomain
+										}
+										siteSlug={ this.props.siteSlug }
+										cancelBundledDomain={ this.state.cancelBundledDomain }
+										purchaseListUrl={ this.props.purchaseListUrl }
+										getConfirmCancelDomainUrlFor={ this.props.getConfirmCancelDomainUrlFor }
+										activeSubscriptions={ this.getActiveMarketplaceSubscriptions() }
+									/>
+
+									<Button
+										className="purchase-cancel-keep-plan"
+										href={ this.props.getManagePurchaseUrlFor(
+											this.props.siteSlug,
+											this.props.purchaseId
+										) }
+									>
+										{ i18n.hasTranslation( 'Keep plan' ) || locale === 'en'
+											? this.props.translate( 'Keep plan' )
+											: this.props.translate( 'Cancel and go back' ) }
+									</Button>
+								</div>
+
+								<p className="cancel-purchase--support-link cancel-purchase--support-link--main">
+									{ this.supportLink() }
+								</p>
+							</Fragment>
+						) }
+
+						{ ! isSubscription( purchase ) && (
+							<Fragment>
+								<CompactCard className="cancel-purchase__product-information">
+									<div className="cancel-purchase__purchase-name">{ purchaseName }</div>
+									<div className="cancel-purchase__description">{ purchaseType( purchase ) }</div>
+									<ProductLink purchase={ purchase } selectedSite={ this.props.site } />
+								</CompactCard>
+
+								<CompactCard className="cancel-purchase__footer">
+									<div className="cancel-purchase__refund-amount">
+										{ this.renderFooterText( this.props ) }
+									</div>
+
+									<CancelPurchaseButton
```

Comment:
Now that this button is in the `! isSubscription()` condition, I'm wondering if it can be simpler. Most of the props passed to it clearly doesn't make sense under this condition, e.g. `cancelBundledDomain`

Reply:
cancelBundledDomain arrived from the props and is mandatory for <CancelPurchaseButton>, I removed the disabled attribute to quickly simplify, let me know if you want me to change the <CancelPurchaseButton> component